fontbe/src/features.rs

```diff
@@ -111,29 +112,42 @@ fn write_debug_fea(context: &Context, is_error: bool, why: &str, fea_content: &s
 impl Work<Context, Error> for FeatureWork {
     fn exec(&self, context: &Context) -> Result<(), Error> {
         let features = context.ir.get_features();
-        if let Features::Empty = *features {
-            // set a default in place so subsequent compiles skip this step
-            trace!("No fea file, dull compile");
-            context.set_features(FontBuilder::default());
-            return Ok(());
-        }
-        let glyph_order = &context.ir.get_final_static_metadata().glyph_order;
-        if glyph_order.is_empty() {
-            warn!("Glyph order is empty; feature compile improbable");
-        }
-        let glyph_map = glyph_order
-            .iter()
-            .map(|n| Into::<FeaRsGlyphName>::into(n.as_str()))
-            .collect();
-
-        let result = self.compile(&features, glyph_map);
-        if result.is_err() || context.flags.contains(Flags::EMIT_DEBUG) {
-            if let Features::Memory(fea_content) = &*features {
-                write_debug_fea(context, result.is_err(), "compile failed", fea_content);
+        if !matches!(*features, Features::Empty) {
+            let glyph_order = &context.ir.get_final_static_metadata().glyph_order;
+            if glyph_order.is_empty() {
+                warn!("Glyph order is empty; feature compile improbable");
+            }
+            let glyph_map = glyph_order
+                .iter()
+                .map(|n| Into::<FeaRsGlyphName>::into(n.as_str()))
+                .collect();
+
+            let result = self.compile(&features, glyph_map);
+            if result.is_err() || context.flags.contains(Flags::EMIT_DEBUG) {
+                if let Features::Memory(fea_content) = &*features {
+                    write_debug_fea(context, result.is_err(), "compile failed", fea_content);
+                }
+            }
+            let buf = result?.build();
+            let font = FontRef::new(&buf).unwrap();
+            debug!(
+                "Built features, gpos? {} gsub? {}",
+                font.gpos().is_ok(),
+                font.gsub().is_ok()
+            );
+            if let Ok(gpos) = font.gpos() {
+                context.set_gpos(gpos.to_owned_table());
             }
+            if let Ok(gsub) = font.gsub() {
+                context.set_gsub(gsub.to_owned_table());
+            }
```

Comment:
this is a very convoluted way of trying to do what we want, I think? Wouldn't it be nicer if we just had some way of retrieving the bytes of a specific table from the `FontBuilder`? (we'd need to add a method there, but nothing crazy)

Reply:
This way it's typed which is nice perhaps? - I agree it feels a bit clunky but ... it does express the intent in fairly straight forward terms

Reply:
Maybe this is a sign that fea-rs should be returning something other than a `FontBuilder`?

It would also be nice if `fea-rs` did not return a type from `write-fonts` directly, since this is a reason why every breaking change in write-fonts is currently a breaking change in fea-rs...

although maybe there's no good work around from this? it certainly doesn't change if I'm returning a set of write-fonts tables instead of a write-fonts TableBuilder. But it still might be a better API.

Reply:
> Maybe this is a sign that fea-rs should be returning something other than a FontBuilder?
>
> It would also be nice if fea-rs did not return a type from write-fonts directly, since this is a reason why every breaking change in write-fonts is currently a breaking change in fea-rs...

Returning any type from write-fonts has the same effect. Raw bytes, or something close to raw bytes (map of tag to raw bytes?) would work. I don't want to have multiple write-fonts in my dependency tree but perhaps it would bumping through versions simpler. Or we could just live with it; I would expect the rate of write-fonts breaks to decline over time as it stabilizes.